packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/common/get-transaction-receipt-by-tx-id.ts

```diff
@@ -0,0 +1,203 @@
+        if (proposalResponsePayload.extension == undefined) continue;
+
+        const responseExtension = proposalResponsePayload.extension;
+        if (responseExtension.chaincode_id == undefined) continue;
+        const extensionChainCodeID = responseExtension.chaincode_id;
+        log.warn("chaincode name and version");
+        //TODO: return chaincode name and version
+        transactReceipt.chainCodeName = extensionChainCodeID;
+        log.warn(extensionChainCodeID.name);
+        transactReceipt.chainCodeName = extensionChainCodeID.name;
+        log.warn(extensionChainCodeID.version);
+        transactReceipt.chainCodeVersion = extensionChainCodeID.version;
+        if (
+          responseExtension.response == undefined ||
+          responseExtension.response.payload == undefined ||
+          responseExtension.response.status == undefined
+        )
+          continue;
+        const responseStatus = responseExtension.response.status;
+        log.warn("responseStatus");
+        log.warn(responseStatus);
+        //TODO: return responseStatus
+        transactReceipt.responseStatus = responseStatus;
+        const extensionResponsePayload = responseExtension.response.payload;
+        log.warn("extensionResponsePayload");
+        log.warn(extensionResponsePayload);
+        //log.warn(JSON.stringify(responseExtension, null, 4));
+        if (
+          responseExtension.results == undefined ||
+          responseExtension.results.ns_rwset == undefined ||
+          responseExtension.results.ns_rwset.length < 2
+        ) {
+          continue;
+        }
+        const extensionNsRwset = responseExtension.results.ns_rwset[1];
+        if (extensionNsRwset.rwset == undefined) continue;
+
+        const rwset = extensionNsRwset.rwset;
+        if (rwset.writes == undefined) continue;
+        const rwsetWrite = rwset.writes;
+        log.warn("rwsetWrite");
+        log.warn(rwsetWrite);
+        if (rwsetWrite[0].key == undefined) continue;
+        const rwsetKey = rwsetWrite[0].key;
+        log.warn("rwsetKey");
+        log.warn(rwsetKey);
+        //TODO: return rwsetKey
+        transactReceipt.rwsetKey = rwsetKey;
+        if (
+          rwsetWrite[0].value == undefined ||
+          rwsetWrite[0].value.data == undefined
+        )
+          continue;
+        const rwSetWriteData = rwsetWrite[0].value.data;
+        log.warn("rwSetWriteData");
+        log.warn(rwSetWriteData);
+        log.warn(typeof rwSetWriteData[0]);
+        // eslint-disable-next-line prefer-spread
+        const rwSetWriteDataStr = String.fromCharCode.apply(
+          String,
+          rwSetWriteData,
+        );
+        log.warn("rwSetWriteDataStr");
+        log.warn(rwSetWriteDataStr);
+        //TODO: return rwSetWriteDataStr
+        transactReceipt.rwsetWriteData = rwSetWriteDataStr;
+        break;
+      }
+    }
+    //TODO: return txIDs
+  }
+  if (block.metadata != undefined && block.metadata.metadata != undefined) {
```

Comment:
@jscode017 It's easier/more robust to check for falsy-ness with the exclamation mark

Reply:
resolved